File: src/main/kotlin/app/kt/gui/ProjectListDialog.java
package app.kt.gui;

import com.intellij.openapi.project.Project;
import com.intellij.openapi.ui.DialogWrapper;
import git4idea.i18n.GitBundle;
import org.jetbrains.annotations.Nullable;

import javax.swing.JComponent;
import javax.swing.JPanel;
import javax.swing.JTable;
import javax.swing.table.DefaultTableModel;
import javax.swing.table.JTableHeader;

public class ProjectListDialog extends DialogWrapper {
    private JPanel mainPanel;
    private JTable mainTable;

    public ProjectListDialog(@Nullable Project project) {
        super(project);
        setTitle("Projects");
    }

    @Nullable
    @Override
    protected JComponent createCenterPanel() {
        return mainPanel;
    }

    private void createUIComponents() {
        DefaultTableModel tableModel = new DefaultTableModel(null, new Object[]{"", "Project", "Branch"});
        mainTable.setModel(tableModel);
    }

}

File: src/main/kotlin/app/gui/ProjectListDialog.kt
package app.gui

import com.intellij.openapi.project.Project
import com.intellij.openapi.ui.DialogWrapper
import com.intellij.ui.table.JBTable
import javax.swing.JComponent
import javax.swing.JPanel
import javax.swing.JTable
import javax.swing.table.DefaultTableModel

class ProjectListDialog(project: Project?) : DialogWrapper(project) {
    private lateinit var mainPanel: JPanel
    private lateinit var mainTable: JTable

    init {
        title = "Projects"
    }

    override fun createCenterPanel(): JComponent? {
        return mainPanel
    }

    private fun createUIComponents() {
        val tableModel = DefaultTableModel(null, arrayOf<Any>("", "Project", "Branch"))

        mainTable = JBTable(tableModel)
    }

}
